chatbot-core/backend/app/databases/qdrant.py

```diff
@@ -0,0 +1,70 @@
+from qdrant_client import QdrantClient, models
+
+from app.settings import Secrets
+from app.utils.logger import LoggerFactory
+
+logger = LoggerFactory().get_logger(__name__)
+
+
+class QdrantConnector:
+    """
+    Qdrant connector class
+
+    Pattern: Singleton
+    Purpose: Create a single instance of the vector database connection
+    """
+
+    _instance = None
+
+    @classmethod
+    def get_instance(cls) -> QdrantClient:
+        """
+        Get the instance of the vector database connection
+        """
+        if cls._instance is None:
+            cls._instance = cls()._create_qdrant_client()
+
+        return cls._instance
+
+    @classmethod
+    def _create_qdrant_client(cls) -> QdrantClient | None:
+        """
+        Create the vector database connection if there is no any existing connection
+        """
+        try:
+            return QdrantClient(host=Secrets.QDRANT_HOST, port=Secrets.QDRANT_PORT)
```

Comment:
In our beta release, it's okay when login to Qdrant without secrets. But in production, I think there should require api key for this.

```python
return QdrantClient(host=Secrets.QDRANT_HOST, port=Secrets.QDRANT_PORT, api_key=Secrets.QDRANT_API_KEY)
```

Reply:
What should be our `api_key` and how to generate it ?

Reply:
The `api_key` is API key for authentication in Qdrant Cloud. Since we use the open source Qdrant with Docker, we don't need it.

Reply:
Fine for now.